src/huntsman/pocs/utils/flats.py

```diff
@@ -0,0 +1,174 @@
+        """
+        if self._n_exposures == 0:
+            return self._initial_exposure_time
+
+        elapsed_time = (current_time() - self._times[-1]).sec
+
+        # Get data for specific camera
+        previous_exptime = self._exptimes[-1]
+        previous_counts = self._average_counts[-1]
+
+        # Calculate next exptime
+        exptime = previous_exptime * (self._target_counts / previous_counts)
+        sky_factor = 2.0 ** (elapsed_time / 180.0)
+        if past_midnight:
+            exptime = exptime / sky_factor
+        else:
+            exptime = exptime * sky_factor
+        exptime = exptime.to_value(u.second) * u.second
+
+        # Make sure the exptime is within limits
+        if exptime >= self._max_exptime:
+            self.logger.warning(f"Truncating exptime at maximum value.")
+            exptime = self._max_exptime
+
+        elif exptime <= self._min_exptime:
+            self.logger.warning(f"Truncating exptime at minimum value.")
+            exptime = self._min_exptime
+
+        return exptime
+
+    def _get_average_counts(self, filename, min_counts=1):
+        """ Read the data and calculate a clipped-mean count rate.
+        Args:
+            filename (str): The filename containing the data.
+            min_counts (float): The minimum count rate returned by this function. Can cause
+                problems if less than or equal to 0, so 1 (default) is a safe choice.
+        Returns:
+            float: The average counts.
+        """
+        try:
+            data = self._load_fits_data(filename)
+        except FileNotFoundError:
+            data = self._load_fits_data(filename + ".fz")
+
+        # Calculate average counts per pixel
+        average_counts, _, _ = sigma_clipped_stats(data - self._bias)
+        if average_counts < min_counts:
+            self.logger.warning('Clipping mean flat-field counts at minimum value: '
+                                f'{average_counts}<{min_counts}.')
+            average_counts = min_counts
+
+        return average_counts
+
+    def _validate_exposure(self, average_counts):
+        """ Check if the previous exposures were either all too faint at the max exposure time, or
+        all too bright at the minimum exposure time.
+        Returns:
+            bool: True if valid, False if not.
+        """
+        return abs(average_counts - self._target_counts) <= self._counts_tolerance
+
+    def _load_fits_data(self, filename, dtype="float32"):
+        """ Load FITS data, using a cutout if necessary.
+        Args:
+            filename (str): The FITS filename.
+            dtype (str or Type): The data type for the returned array.
+        Returns:
+            np.array: The exposure data.
+        """
+        data = fits.getdata(filename)
+        if self._cutout_size is not None:
+            x, y = data.shape[1]/2, data.shape[0]/2
+            data = Cutout2D(data, (x, y), size=self._cutout_size).data
+        return data.astype(dtype)
```

Comment:
This can be simplified using existing tools:

```suggestion
        data = fits_utils.getdata(filename).astype(dtype)
        if self._cutout_size is not None:
            data = crop_data(data, box_size=self._cutout_size)
            
        return data
```